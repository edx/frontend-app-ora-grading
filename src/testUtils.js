// eslint-disable-next-line import/prefer-default-export
export const formatMessage = (msg, values) => {
  let message = msg.defaultMessage;
  if (values === undefined) { return message; }
  Object.keys(values).forEach(key => {
    // eslint-disable-next-line
    message = message.replace(`{${key}}`, values[key]);
  });
  return message;
};
