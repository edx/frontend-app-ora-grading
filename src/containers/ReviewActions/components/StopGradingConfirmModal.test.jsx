import { shallow } from 'enzyme';

import { formatMessage } from 'testUtils';
import { StopGradingConfirmModal } from './StopGradingConfirmModal';

jest.mock('components/ConfirmModal', () => 'ConfirmModal');

describe('StopGradingConfirmModal', () => {
  const props = {
    intl: { formatMessage },
    isOpen: false,
    isOverride: false,
    onCancel: jest.fn().mockName('this.props.onCancel'),
    onConfirm: jest.fn().mockName('this.props.onConfirm'),
  };
  test('snapshot: closed', () => {
    expect(shallow(<StopGradingConfirmModal {...props} />)).toMatchSnapshot();
  });
  test('snapshot: open', () => {
    expect(shallow(<StopGradingConfirmModal {...props} isOpen />)).toMatchSnapshot();
  });
  test('snapshot: open, isOverride', () => {
    expect(shallow(<StopGradingConfirmModal {...props} isOverride />)).toMatchSnapshot();
  });
});
