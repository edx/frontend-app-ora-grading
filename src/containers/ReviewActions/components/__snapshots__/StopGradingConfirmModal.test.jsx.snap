// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopGradingConfirmModal snapshot: closed 1`] = `undefined`;

exports[`StopGradingConfirmModal snapshot: open 1`] = `undefined`;

exports[`StopGradingConfirmModal snapshot: open, isOverride 1`] = `undefined`;
